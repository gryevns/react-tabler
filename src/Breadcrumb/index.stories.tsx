import React from "react";

import Breadcrumb from "../Breadcrumb";

export default {
    title: "Breadcrumb",
};

const types = [undefined, "dots" as const, "arrows" as const, "bullets" as const];

export const Types = () =>
    types.map(type => (
        <Breadcrumb type={type}>
            <Breadcrumb.Item>
                <a href="#">Home</a>
            </Breadcrumb.Item>
            <Breadcrumb.Item>
                <a href="#">Library</a>
            </Breadcrumb.Item>
            <Breadcrumb.Item active>
                <a href="#">Data</a>
            </Breadcrumb.Item>
        </Breadcrumb>
    ));

export const Alternate = () =>
    types.map(type => (
        <Breadcrumb alternate type={type}>
            <Breadcrumb.Item>
                <a href="#">Home</a>
            </Breadcrumb.Item>
            <Breadcrumb.Item>
                <a href="#">Library</a>
            </Breadcrumb.Item>
            <Breadcrumb.Item active>
                <a href="#">Data</a>
            </Breadcrumb.Item>
        </Breadcrumb>
    ));
