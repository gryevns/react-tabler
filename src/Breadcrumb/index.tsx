import React from "react";
import cn from "classnames";

interface BreadcrumbItemProps {
    children: React.ReactNode;
    className?: string;
    active?: boolean;
}

const BreadcrumbItem = ({ children, className, active }: BreadcrumbItemProps) => {
    const classes = cn(`breadcrumb-item`, { active }, className);
    const props = {};
    if (active) {
        props["aria-current"] = "page";
    }
    return (
        <li className={classes} {...props}>
            {children}
        </li>
    );
};

type Type = "dots" | "arrows" | "bullets";

interface BreadcrumbProps {
    children: React.ReactNode;
    className?: string;
    type?: Type;
    alternate?: boolean;
}

export const Breadcrumb = ({
    children,
    className,
    type,
    alternate = false,
}: BreadcrumbProps) => {
    const classes = cn(
        `breadcrumb`,
        {
            "breadcrumb-dots": type === "dots",
            "breadcrumb-arrows": type === "arrows",
            "breadcrumb-bullets": type === "bullets",
            "breadcrumb-alternate": alternate,
        },
        className,
    );
    return (
        <ol className={classes} aria-label="breadcrumbs">
            {children}
        </ol>
    );
};

Breadcrumb.Item = BreadcrumbItem;

export default Breadcrumb;
