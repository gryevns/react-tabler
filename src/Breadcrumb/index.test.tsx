import React from "react";
import { shallow } from "enzyme";

import Breadcrumb, { Type } from "./index";

describe("Breadcrumb", () => {
    it("should match types snapshot", () => {
        [undefined, "dots", "arrows", "bullets"].map((type: Type) => {
            const wrapper = shallow(
                <Breadcrumb type={type}>
                    <Breadcrumb.Item>
                        <a href="#">Home</a>
                    </Breadcrumb.Item>
                    <Breadcrumb.Item>
                        <a href="#">Library</a>
                    </Breadcrumb.Item>
                </Breadcrumb>,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
