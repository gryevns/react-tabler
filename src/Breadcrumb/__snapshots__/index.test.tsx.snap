// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should match types snapshot 1`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb"
>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Home
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Library
    </a>
  </BreadcrumbItem>
</ol>
`;

exports[`Breadcrumb should match types snapshot 2`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb breadcrumb-dots"
>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Home
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Library
    </a>
  </BreadcrumbItem>
</ol>
`;

exports[`Breadcrumb should match types snapshot 3`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb breadcrumb-arrows"
>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Home
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Library
    </a>
  </BreadcrumbItem>
</ol>
`;

exports[`Breadcrumb should match types snapshot 4`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb breadcrumb-bullets"
>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Home
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem>
    <a
      href="#"
    >
      Library
    </a>
  </BreadcrumbItem>
</ol>
`;
